src/shared/libfido2-util.c

```diff
@@ -776,10 +777,19 @@ int fido2_generate_hmac_hash(
         if (!c)
                 return log_oom();
 
-        r = sym_fido_cred_set_extensions(c, FIDO_EXT_HMAC_SECRET);
+        int extensions = FIDO_EXT_HMAC_SECRET;
+        if (FLAGS_SET(lock_with, FIDO2ENROLL_PIN)) {
+                extensions |= FIDO_EXT_CRED_PROTECT;
+
+                r = sym_fido_cred_set_prot(c, FIDO_CRED_PROT_UV_REQUIRED);
+                if (r != FIDO_OK)
+                        log_notice("Failed to set protection level on FIDO2 credential: %s", sym_fido_strerr(r));
+        }
+
+        r = sym_fido_cred_set_extensions(c, extensions);
```

Comment:
hmm, isn't this going to fail if FIDO_EXT_CRED_PROTECT is not supported?

Reply:
I'll take a look at what libfido2 does, but on the authenticator (CTAP) side, what should happen is that the extension is ignored and the credential created without error. The result from the makeCredential wouldn't have credProtect in it.

All extensions are this way in CTAP - ignored by authenticators that don't support them.

Reply:
https://github.com/Yubico/libfido2/blob/b63570ab677354cddee60075f522331fb1a66b0a/src/cred.c#L946 - no failure on the set_extensions call.

https://github.com/Yubico/libfido2/blob/b63570ab677354cddee60075f522331fb1a66b0a/src/cred.c#L91 - unconditionally shoved into the makeCredential payload by the library. ( https://github.com/Yubico/libfido2/blob/b63570ab677354cddee60075f522331fb1a66b0a/src/cbor.c#L627 no error on the encoding).

Read of the code says it should be okay. I'll test it end to end later today.

Reply:
can you summarize this in a short comment in the code here. i.e. just mention that tokens are supposed to gracefully ignore our request to use the extension if they dont support it.

Reply:
btw, a comment explaining briefly what use of the extension is supposed to achieve would be good anyway. i.e. some of the commit msg text should probably be a comment.

Reply:
I've added comments describing the logic. Please let me know if they're not sufficiently clear.